src/transformState.js

```diff
@@ -62,7 +62,56 @@
  * @param {Object[]} transforms
  */
 function transformState(state, transforms) {
-  // write code here
+  for (let i = 0; i < transforms.length; i++) {
+    const { operation, properties } = transforms[i];
+
+    switch (operation) {
+      case ('addProperties'):
+        for (const key in properties) {
+          state[key] = properties[key];
+        }
+        break;
+
+      case ('removeProperties'):
+        for (let k = 0; k < properties.length; k++) {
+          if (state.hasOwnProperty(properties[k])) {
+            delete state[properties[k]];
+          }
+        }
+        break;
+
+      case ('clear'):
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+      default:
+        break;
+    }
+    // 'SOLUTION WITH 'if':
+
+    // if (transforms[i].operation === 'addProperties') {
+    //   for (const key in transforms[i].properties) {
+    //     state[key] = transforms[i].properties[key];
+    //   }
+    // }
+
+    // if (transforms[i].operation === 'removeProperties') {
+    //   for (let k = 0; k < transforms[i].properties.length; k++) {
+    //     if (state.hasOwnProperty(transforms[i].properties[k])) {
+    //       delete state[transforms[i].properties[k]];
+    //     }
+    //   }
+    // }
+
+    // if (transforms[i].operation === `clear`) {
+    //   for (const key in state) {
+    //     delete state[key];
+    //   }
+    // }
```

Comment:
no need to save this, the switch case is better

Reply:
deleted comments